typescript/test/data/redemption.ts

```diff
@@ -666,3 +667,123 @@ export const redemptionProof: RedemptionProofTestData = {
       "03989d253b17a6a0f41838b84ff0d20e8898f9d7b1a98f2564da4cc29dcf8581d9",
   },
 }
+
+export const findWalletForRedemptionData: {
+  newWalletRegisteredEvents: NewWalletRegisteredEvent[]
+  wallets: {
+    [walletPublicKeyHash: string]: {
+      state: WalletState
+      mainUtxoHash: Hex
+      walletPublicKey: Hex
+      btcAddress: string
+      utxos: UnspentTransactionOutput[]
+    }
+  }
+} = {
+  newWalletRegisteredEvents: [
+    {
+      blockNumber: 8367602,
+      blockHash: Hex.from(
+        "0x908ea9c82b388a760e6dd070522e5421d88b8931fbac6702119f9e9a483dd022"
+      ),
+      transactionHash: Hex.from(
+        "0xc1e995d0ac451cc9ffc9d43f105eddbaf2eb45ea57a61074a84fc022ecf5bda9"
+      ),
+      ecdsaWalletID: Hex.from(
+        "0x5314e0e5a62b173f52ea424958e5bc04bd77e2159478934a89d4fa193c7b3b72"
+      ),
+      walletPublicKeyHash: Hex.from(
+        "0x03b74d6893ad46dfdd01b9e0e3b3385f4fce2d1e"
+      ),
+    },
+    {
+      blockNumber: 8502240,
+      blockHash: Hex.from(
+        "0x4baab7520cf79a05f22723688bcd1f2805778829aa4362250b8ee702f34f4daf"
+      ),
+      transactionHash: Hex.from(
+        "0xe88761c7203335e237366ec2ffca1e7cf2690eab343ad700e6a6e6dc236638b1"
+      ),
+      ecdsaWalletID: Hex.from(
+        "0x0c70f262eaff2cdaaddb5a5e4ecfdda6edad7f1789954ad287bfa7e594173c64"
+      ),
+      walletPublicKeyHash: Hex.from(
+        "0x7670343fc00ccc2d0cd65360e6ad400697ea0fed"
+      ),
+    },
+    {
+      blockNumber: 8981644,
+      blockHash: Hex.from(
+        "0x6681b1bb168fb86755c2a796169cb0e06949caac9fc7145d527d94d5209a64ad"
+      ),
+      transactionHash: Hex.from(
+        "0xea3a8853c658145c95165d7847152aeedc3ff29406ec263abfc9b1436402b7b7"
+      ),
+      ecdsaWalletID: Hex.from(
+        "0x7a1437d67f49adfd44e03ddc85be0f6988715d7c39dfb0ca9780f1a88bcdca25"
+      ),
+      walletPublicKeyHash: Hex.from(
+        "0x328d992e5f5b71de51a1b40fcc4056b99a88a647"
+      ),
+    },
+  ],
+  wallets: {
```

Comment:
Can we add some wallets that should be skipped by the wallet selection? E.g. some non-live state, not enough utxo value, main utxo not set. 
These non-matching cases should be on top of the list to see that the mechanism skips them and finds the first wallet that matches all the criteria.

Reply:
24ede08